src/Compilers/CSharp/Portable/FlowAnalysis/ControlFlowPass.cs

```diff
@@ -323,7 +323,6 @@ public override BoundNode VisitSwitchSection(BoundSwitchSection node, bool lastS
                 var boundLabel = node.SwitchLabels.Last();
                 Diagnostics.Add(lastSection ? ErrorCode.ERR_SwitchFallOut : ErrorCode.ERR_SwitchFallThrough,
                                 new SourceLocation(boundLabel.Syntax), boundLabel.Label.Name);
-                this.State.Reported = true;
```

Comment:
The removal looks OK, but I'm trying to understand why this was here in the first place.

Was the intention that switch fall-through was considered unreachable code? I can see how it may be interpreted that way. And then the assert fired because the local function is still reachable?

Reply:
The idea of the `Reported` state was that once we reported an error for unreachable code (the principal error reported in the control flow pass), we don't need to report it on subsequent statements. We used the `Reported` bit to record the fact that the unreachable state was already reported as an error and therefore does not deserve another error. So the pattern was to set `Reported` whenever reporting the error.

But that pattern was followed too slavishly: we report an error here, but it is not an error of unreachable code (on the contrary, it is an error of *reachable* code where it should not be reachable). The assertion is that a state is either not reachable or not reported (because if it were reachable it should not have been reported). But setting that bit here clearly violates the invariant (see the enclosing `if`).

In short, the code being deleted here was simple a thinko.

Reply:
OK, so the state here is reachable. This matches my expectations.